core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -294,6 +299,7 @@ protected void startServingRPCServer() {
 
       // Wait until the server is shut down.
       mGrpcServer.awaitTermination();
+      executorService.shutdown();
```

Comment:
Could this block unnecessarily or are all threads cleaned up in the `awaitTermination` above?

Reply:
Once awaitTermination will return as soon as mGrpcServer.shutdown() is called so there could be tasks that are still active in the pool.
Moving pool termination to stopServing() could be a better option.

Reply:
my understanding is that `shutdown` simply stops the pool from taking in more tasks, it does not block to wait for completion of  tasks.   so maybe we are ok?

Reply:
In the stopServing(), after shutting down the grpc server. We can call shutdown() and shutdownNow() immediately after the each other.

Reply:
done. PTAL